packages/commonwealth/server/routes/getNewProfile.ts

```diff
@@ -0,0 +1,92 @@
+import type { Request, Response, NextFunction } from 'express';
+import { Op } from 'sequelize';
+import type { DB } from '../models';
+
+export const Errors = {
+  NoAddressProvided: 'No address provided in query',
+  NoAddressFound: 'No address found',
+  NoProfileFound: 'No profile found',
+};
+
+const getNewProfile = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const { address } = req.query;
+  if (!address) return next(new Error(Errors.NoAddressProvided));
+
+  const addressModel = await models.Address.findOne({
+    where: {
+      address,
+    },
+    include: [models.Profile],
+  });
+  if (!addressModel) return next(new Error(Errors.NoAddressFound));
+
+  const profile = await models.Profile.findOne({
+    where: {
+      id: addressModel.profile_id,
+    },
+  });
+  if (!profile) return next(new Error(Errors.NoProfileFound));
+
+  const addresses = await models.Address.findAll({
+    where: {
+      profile_id: profile.id,
+    },
+  });
+
+  const chainIds = [...new Set<string>(addresses.map((a) => a.chain))];
+  const chains = await models.Chain.findAll({
```

Comment:
Again, we can use `include` here to reduce the number of queries made.

However, I want to note that Address should not be the source of truth for "chains", which I assume is used for "membership in communities". That should instead be fetched via the RoleAssignments on the Address objects, which refer to CommunityRoles that have Chains.

Ideally we should make a helper for this query, as it uses some nested includes. You can see an example of it here: https://github.com/hicommonwealth/commonwealth/pull/2810/files#diff-325387b799e25c792b96772560f6cf3af30ec8eeb67b9b9dced6fd62c6c2de44R23

Reply:
After double checking why I need chains here, it looks like I can actually remove it. I only needed chain info to get the `iconUrl` but realized I can just get it from `app.config.chains.getById(chain).iconUrl`.